common/src/main/java/io/pravega/common/util/TypedProperties.java

```diff
@@ -125,27 +125,35 @@ public boolean getBoolean(Property<Boolean> property) throws ConfigurationExcept
     }
 
     private <T> T tryGet(Property<T> property, Function<String, T> converter) {
-        String fullKeyName = this.keyPrefix + property.getName();
+        String propNewName = this.keyPrefix + property.getName();
+        String propOldName = this.keyPrefix + property.getLegacyName();
+        String propValue = null;
 
-        // Get value from config.
-        String value = this.properties.getProperty(fullKeyName, null);
+        if (property.hasLegacyName()) {
+            // Value of property with old name
+            propValue = this.properties.getProperty(propOldName, null);
+        }
 
-        if (value == null && property.hasLegacyName()) {
-            value = this.properties.getProperty(this.keyPrefix + property.getLegacyName(), null);
+        if (propValue == null) {
+            // Value of property with new name
+            propValue = this.properties.getProperty(propNewName, null);
         }
-        if (value == null) {
-            // 2. Nothing in the configuration for this Property.
+
+        // A property with neither old not new name was defined, so the property value is still null
+        if (propValue == null) {
             if (property.hasDefaultValue()) {
                 return property.getDefaultValue();
```

Comment:
Also call out in log that default value is being used.

Reply:
No, there is no need to pollute the logs with this information. Both the Segment store and controller log the values so you should be able to see them if you parse the logs. 

No point in adding noise to the logs.